utilities/operator.py

```diff
@@ -730,16 +713,19 @@ def get_generated_icsp_idms(
     pull_secret_directory: str,
     is_idms_cluster: bool,
     cnv_version: str | None = None,
+    filter_options: str = "",
 ) -> str:
     pull_secret = None
     if BREW_REGISTERY_SOURCE in image_url:
         registry_source = BREW_REGISTERY_SOURCE
+    if registry_source == BREW_REGISTERY_SOURCE:
```

Comment:
not needed. as long as image url shows it is brew image, irrespective of anything, you need to set pull secret.

Reply:
The change keeps the same way of adding the `pull_secret` before.
I am worried about upgrade tests where the source might be not a BREW_REGISTRY, if the source isn't a BREW_REGISTRY will it still work with an addition `generated_pulled_secret` ?

Reply:
The existing code works for upgrade, your change is not needed.

Reply:
What do you mean?
It is needed for descheduler image.
The descheduler image does not use an image URL with `BREW_REGISTERY_SOURCE`, it's URL is `quay.io`, therefore in the existing code, it set `pull_secret = None`.

This change is needed so we will use the pull_secret even if we don't use a BREW_REGISTRY.

An alternative might be always pass `generated_pulled_secret`, and insert a logic of `if BREW_IMAGE_SOURCE` before the function